tests/test_api.py

```diff
@@ -52,4 +52,8 @@ def test_waveform():
     signal = np.random.randn(2000)
     for n in [400, 800, 1200]:
         audplot.waveform(signal[:n])
+        ax = plt.gca()
+        xlim = ax.get_xlim()
+        assert xlim[0] == 0
+        assert xlim[1] == n
```

Comment:
This currently only works because you have the case `n < pixels`. With a larger `n` it fails, but I think that is only because of the following line you added: `samples = x_axis[-1] + 1`. When I removed it, then the test was also passing for larger values of `n`.

Reply:
This is strange, I checked locally, and for me it has `pixels=800`, which means the 1200 samples test case is already > pixels.